### PR TITLE
Add issue templates

### PR DESCRIPTION
**Pull request by Perflyst**

### CODE REVIEW

**Comment by Perflyst** [comment]
This could help to avoid duplicated reports, part of https://github.com/iv-org/invidious/issues/1684

**Comment by Perflyst** [comment]
@TheFrenchGhosty @hackerncoder what do you think?

**Review by TheFrenchGhosty — COMMENTED** [review]
Feature requests and Enhancements should be 2 different templates.

Enhancements should have the "enhancement" label.

Feature Requests should have the "feature-request" label (I just created it), and their title should start with "[Feature Request]".

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+---
+name: Bug report
+about: Create a report to help us improve
```
Comment (TheFrenchGhosty) [inline .github/ISSUE_TEMPLATE/bug_report.md:3]: "Create a bug report to help us improve Invidious."

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+title: ''
+labels: bug
+assignees: ''
+
+---
+
+<!-- Please use the search function to check if the bug you found has already been reported by someone else -->
+<!-- If you want to suggest a new feature please use "Feature request" instead -->
```
Comment (TheFrenchGhosty) [inline .github/ISSUE_TEMPLATE/bug_report.md:11]: ``` <!-- If you want to suggest a new feature please use "Feature request" instead --> ```
``` <!-- If you want to suggest an enhancement to a new feature please use "Enhancement" instead --> ```

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+**Steps to Reproduce**
+<!-- Steps to reproduce the behavior:
+1. Go to '...'
+2. Click on '....'
+3. Scroll down to '....'
+4. See error
+-->
+
+**Logs**
+<!-- If applicable, add logs where the error is reported. -->
```
Comment (TheFrenchGhosty) [inline .github/ISSUE_TEMPLATE/bug_report.md:25]: ``` <!-- If applicable, copy the log that appear in the browser page where the error is reported. --> ```

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+
+**Screenshots**
+<!-- If applicable, add screenshots to help explain your problem. -->
+
+**Additional context**
+<!-- Add any other context about the problem here. -->
```
Comment (TheFrenchGhosty) [inline .github/ISSUE_TEMPLATE/bug_report.md:31]: ```Web browser (if applicable):``` 
```OS (if applicable):```

File: .github/ISSUE_TEMPLATE/feature_request.md [hunk 5]
```diff
@@ -0,0 +1,23 @@
+---
+name: Feature request
+about: Suggest an idea for this project
+title: ''
```
Comment (TheFrenchGhosty) [inline .github/ISSUE_TEMPLATE/feature_request.md:4]: ```title: '[Feature Request] YourTitle'```

File: .github/ISSUE_TEMPLATE/feature_request.md [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+**Is your feature request related to a problem? Please describe.**
+<!-- A clear and concise description of what the problem is. Ex. I'm always frustrated when [...] -->
+
+**Describe the solution you'd like**
+<!-- A clear and concise description of what you want to happen. -->
+
+**Describe alternatives you've considered**
+<!-- A clear and concise description of any alternative solutions or features you've considered. -->
+
+**Additional context**
```
Comment (TheFrenchGhosty) [inline .github/ISSUE_TEMPLATE/feature_request.md:22]: "If you know the part of the code that need to be enhanced, please link it here"

**Review by TheFrenchGhosty — CHANGES_REQUESTED** [review]

**Comment by Perflyst** [comment]
> Feature requests and Enhancements should be 2 different templates.

What's the difference?

**Comment by TheFrenchGhosty** [comment]
@Perflyst Enhancement enhance an existing feature, Feature Request add a new feature entirely.

File: .github/ISSUE_TEMPLATE/bug_report.md (hunk 2 above)
Comment (Perflyst) [reply .github/ISSUE_TEMPLATE/bug_report.md:11]: I guess you wanted to write something like
` <!-- If you want to suggest an enhancement to an existing feature please use "Enhancement" instead -->`?

**Comment by Perflyst** [comment]
An enhancement to an existing feature is also a feature request IMHO. Additionally we would need to re-tag all feature requests. I would prefer using enhancement as is for both

File: .github/ISSUE_TEMPLATE/feature_request.md (hunk 6 above)
Comment (Perflyst) [reply .github/ISSUE_TEMPLATE/feature_request.md:22]: Most won't know the code and if people know the code they automatically will link it, that's what already happened in the past without any template. This additional sentence is just more reading for the normal user.

File: .github/ISSUE_TEMPLATE/bug_report.md (hunk 2 above)
Comment (TheFrenchGhosty) [reply .github/ISSUE_TEMPLATE/bug_report.md:11]: Yes, that was a typo.

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 7]
```diff
@@ -1,14 +1,15 @@
 ---
 name: Bug report
-about: Create a report to help us improve
-title: ''
+about: Create a report to help us improve Invidious
```
Comment (TheFrenchGhosty) [inline .github/ISSUE_TEMPLATE/bug_report.md:3]: Should be:

Create a **bug** report to help us improve Invidious

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 8]
```diff
@@ -1,14 +1,15 @@
...
+title: '[Bug] '
 labels: bug
 assignees: ''
 
 ---
 
 <!-- Please use the search function to check if the bug you found has already been reported by someone else -->
 <!-- If you want to suggest a new feature please use "Feature request" instead -->
+<!-- If you want to suggest an enhancement to a new feature please use "Enhancement" instead -->
```
Comment (TheFrenchGhosty) [inline .github/ISSUE_TEMPLATE/bug_report.md:12]: >If you want to suggest an enhancement to a new feature please use "Enhancement" instead

Should be: 

If you want to suggest an enhancement to an **existing** feature please use "Enhancement" instead

**Review by TheFrenchGhosty — CHANGES_REQUESTED** [review]

**Comment by TheFrenchGhosty** [comment]
>An enhancement to an existing feature is also a feature request IMHO. Additionally we would need to re-tag all feature requests. I would prefer using enhancement as is for both

Yes, but an enhancement is usually "easier" to implement that a completely new feature. Reason why I think both should be separated.

File: .github/ISSUE_TEMPLATE/feature_request.md (hunk 6 above)
Comment (TheFrenchGhosty) [reply .github/ISSUE_TEMPLATE/feature_request.md:22]: Agree.

File: .github/ISSUE_TEMPLATE/feature_request.md [hunk 9]
```diff
@@ -0,0 +1,23 @@
+---
+name: Feature request
+about: Suggest an idea for this project
+title: '[Feature request] '
+labels: enhancement
```
Comment (hackerncoder) [inline .github/ISSUE_TEMPLATE/feature_request.md:5]: If it is a feature request it should have the feature-request label. Of course, this goes together with splitting enhancement and feature-request.

**Comment by Perflyst** [comment]
I am going to split feature request and enhancement then

**Comment by Perflyst** [comment]
Feature request and enhancement is now separated. Please check again

**Review by hackerncoder — APPROVED** [review]

**Review by TheFrenchGhosty — APPROVED** [review]

**Pull request merged**